scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Connection.java

```diff
@@ -283,7 +292,62 @@ public boolean addInbound(Record record) {
 	 * 
 	 * @return the next record or {@code null} if the buffer is empty.
 	 */
-	public Record getNextInboundRecord() {
-		return incoming.poll();
+	private Record getNextInboundRecord() {
+
+		synchronized(incoming) {
```

Comment:
I guess it will do not harm then, right? If the monitor is already acquired, no waiting is needed or is it?